File: src/main/java/com/cuupa/classificator/services/kb/TokenTextPointer.java
package com.cuupa.classificator.services.kb;

import java.util.Arrays;

public class TokenTextPointer {

	private char[] charArray;
	private int index;
	
	public TokenTextPointer(char[] charArray, int index) {
		this.charArray = Arrays.copyOf(charArray, charArray.length);
		this.index = index;
	}

	public int getIndex() {
		return index;
	}

	public char get(int i) {
		return charArray[i];
	}

	public int getCharSize() {
		return charArray.length;
	}
}

File: engine/src/main/kotlin/com/cuupa/classificator/engine/services/token/TokenTextPointer.kt
package com.cuupa.classificator.engine.services.token

class TokenTextPointer(charArray: CharArray, val index: Int) {

    private val charArrayIntern: CharArray = charArray.copyOf(charArray.size)

    operator fun get(i: Int): Char {
        return charArrayIntern[i]
    }

    val charSize: Int
        get() = charArrayIntern.size

}